### DISCUSSION TITLE
BelongsTo relationship returning 30k+ results when editing a resource with relationship already set

### DISCUSSION THREAD

**Discussion by Livijn** [laravel/nova-issues#5013] [Q&A]
- Laravel Version: v9.36.2
- Nova Version: 4.16.1
- PHP Version: 8.1.11
- Database Driver & Version: MySQL 8
- Operating System and Version: MacOS
- Browser type and version: Chrome 106.0.5249.119
- Reproduction Repository: none

### Description:
BelongsTo relationship returning 30k+ results when editing a resource with relationship already set. Also, when creating a resource, it returns tons of results.

**Query params:**
```
current: 19239
first: true
search: 
withTrashed: false
resourceId: 829
viaResource: 
viaResourceId: 
viaRelationship: 
editing: true
editMode: update
```

I know I can limit the results with `$relatableSearchResults`, but then the field is just empty. I have upgraded from Laravel 3, not sure if that's related.

### Detailed steps to reproduce the issue on a fresh Nova installation:
Create a BelongsTo relationship with 30k+ rows.

**Comment by crynobone** [comment]
> BelongsTo relationship returning 30k+ results when editing a resource with relationship already set.

Other than accessing from related viaResource, you are still able to change the selected resource hence it is required to load all other options. You should use `searchable()` in this scenario.

**Reply by Livijn** [reply]
I am already using `searchable()`, that's where the query params are from.

**Reply by crynobone** [reply]
Sorry, I was not able to replicate the issue with `searchable()`. Query string with `current` and `first: true` should already return a single resource unless you have `relatableQuery()` appending the query to cause it to return 30+k resources.
